lesson22/app.js

```diff
@@ -1,225 +1,235 @@
-  const tabletHeaderArray = Array.from(tabletHeader);
-  const sortsDataAttributes = [];
-  tabletHeaderArray.forEach((tabletHeader) => {
-    if (tabletHeader.hasAttribute("data-sort")) {
-      sortsDataAttributes.push(tabletHeader);
-    }
-  });
-  createSortButtons(sortsDataAttributes);
-}
-
-function createSortButtons(sortsDataAttributes) {
-  const ascButton = buttonAdd("ASC", sortsDataAttributes);
-  const descButton = buttonAdd("DESC", sortsDataAttributes);
-
-  sortsDataAttributes.forEach((sortsDataAttribute, index) => {
-    const sortButtonContainerDiv = document.createElement("div");
-    sortButtonContainerDiv.classList.add("sort-button_container");
-    sortButtonContainerDiv.appendChild(ascButton[index]);
-    sortButtonContainerDiv.appendChild(descButton[index]);
-    sortsDataAttribute.appendChild(sortButtonContainerDiv);
-  });
-}
-
-function buttonAdd(buttonType, sortsDataAttributes) {
-  const buttons = [];
-  sortsDataAttributes.forEach((value, index) => {
-    const button = document.createElement("button");
-    button.id = `${buttonType}_${sortsDataAttributes[index].id}`;
-    button.classList.add("sort-button", `_${buttonType}`);
-    if (buttonType === "ASC") {
-      button.textContent = "▲";
-    } else {
-      button.textContent = "▼";
-    }
-    button.dataset.status = `${buttonType}`;
-    button.dataset.type = `${sortsDataAttributes[index].id}`;
-    buttons.push(button);
-  });
-  return buttons;
-}
+const sortDescFunctor = (key) => (a, b) => {
+  if (a[key] > b[key]) return -1;
+  if (a[key] < b[key]) return 1;
+  return 0;
+};
 
-function sortAsc(data, buttonType) {
-  const sortResult = [...data].sort(function (a, b) {
-    return a[buttonType] - b[buttonType];
-  });
-  createTableBody(sortResult);
-}
+const sortAscFunctor = (key) => (a, b) => {
+  if (a[key] < b[key]) return -1;
+  if (a[key] > b[key]) return 1;
+  return 0;
+};
 
-function sortDesc(data, buttonType) {
-  const sortResult = [...data].sort(function (a, b) {
-    return b[buttonType] - a[buttonType];
+const renderSortingFailedMessage = () => {
+  const tableElem = document.getElementById("js-table");
+  parent.removeChild(tableElem);
+  const message = createAttributedElements(
+    "p",
+    { class: "sorting-failed_text" },
+    "並び替えすることが出来ませんでした"
+  );
+  parent.appendChild(message);
+  return false;
+};
+
+const sortDesc = (target, key) => {
+  if (target === null) return renderSortingFailedMessage();
+  const sortTarget = [...target];
+  sortTarget.sort(sortDescFunctor(key));
+  return sortTarget;
+};
+
+const sortAsc = (target, key) => {
+  if (target === null) return renderSortingFailedMessage();
+  const sortTarget = [...target];
+  sortTarget.sort(sortAscFunctor(key));
+  return sortTarget;
+};
+
+const initExceptTargetButtons = (target) => {
+  const sortButtons = [...document.querySelectorAll(".button")];
+  sortButtons.forEach((value) => {
+    if (value === target) return;
```

Comment:
IMO
I think `button` would be more appropriate.
```suggestion
  sortButtons.forEach((button) => {
    if (value === target) return;
```

Reply:
Thanks for the comment.
You are absolutely right. Corrected.
[f089fd8](https://github.com/443n0511/js-lesson/pull/44/commits/f089fd8023c001637ef46002180ee97b6a9cb67f)